portfolio/src/main/webapp/script.js

```diff
@@ -15,30 +20,163 @@ function plusSlides(nextSlide, slideSection) {
 // nextSlide represents whether the slide will switch to the next or previous slide
 // slideSection represents which section of slides is receving the change as referenced by "slideSections"
 function showSlides(nextSlide, slideSection) {
-  var i;
-  var slides = document.getElementsByClassName(slideSections[slideSection]);
-  if (nextSlide> slides.length) {slideIndices[slideSection] = 1}    
-  if (nextSlide < 1) {slideIndices[slideSection] = slides.length}
-  for (i = 0; i < slides.length; i++) {
-      slides[i].style.display = "none";  
-  }
-    slides[slideIndices[slideSection]-1].style.display = "block";  
-}
-
-function getComments() {
-   fetch('/data').then(response => response.json()).then((commentInfo) => {
-    const commentBox = document.getElementById("comment-box");
-    const numComments = commentInfo.numComments;
-    const comments = commentInfo.commentList;
+    var i;
+    var slides = document.getElementsByClassName(slideSections[slideSection]);
+
+    if (slideSection > 0) {
+        modals = document.getElementsByClassName(modalBoxes[slideSection - 1]);
+    }
+
+    if (nextSlide > slides.length) {
+        slideIndices[slideSection] = 1;
+    }
+    if (nextSlide < 1) {
```

Comment:
I am not sure I understand what you are trying to do here. Are you trying to wrap the value to be within 1 and slide.length?  Why not clip it instead of wrapping?  This may also be an appropriate place to add a comment to indicate your intention.

By the way, if you wanted to clip then you can achieve that with a single statement:

slideIndices[slideSection] = Math.min(Math.max(nextSlide, 1), slides.length)

If not then I recommend replacing the second 'if' by 'else if'.

Reply:
Wrapping is a bit tricky, because all values of nextSlide > slide.length wrap to 1.  Similarly, all values less than 1 (only 0, since the values cannot be negative?) wrap to slide.length. Clipping might be more intuitive.

In any case, I suggest adding a comment explaining what you are trying to do.

Reply:
I added a comment in. I'm not sure if clipping the value would still preserve my functionality, I'll revisit how I could able implement it.